File: musiclib/src/main/java/com/music/lake/musiclib/player/BasePlayer.java
package com.music.lake.musiclib.player;

import com.music.lake.musiclib.bean.BaseMusicInfo;
import com.music.lake.musiclib.playback.PlaybackListener;

public class BasePlayer {
    public BaseMusicInfo mNowPlayingMusic;
    public PlaybackListener listener;
    /**
     * 准备好了直接播放
     */
    public boolean playWhenReady;

    public void stop() {
    }

    public void release() {
    }

    public void start() {
    }

    public void pause() {
    }

    public boolean isPlaying() {
        return false;
    }

    public long position() {
        return 0;
    }

    public long duration() {
        return 0;
    }

    /**
     * 获取标题
     */
    public String getTitle() {
        if (mNowPlayingMusic != null) {
            return mNowPlayingMusic.getTitle();
        }
        return "";
    }

    /**
     * 获取歌手专辑
     *
     * @return
     */
    public String getArtistName() {
        if (mNowPlayingMusic != null) {
            return mNowPlayingMusic.getArtist();
        }
        return null;
    }

    public boolean isInitialized() {
        return true;
    }

    public void seekTo(long ms) {
    }

    public boolean isPrepared() {
        return false;
    }

    public int bufferedPercentage() {
        return 0;
    }

    public int getAudioSessionId() {
        return 0;
    }

    public void setDataSource(String uri) {
    }

    public void setMusicInfo(BaseMusicInfo mNowPlayingMusic) {
        this.mNowPlayingMusic = mNowPlayingMusic;
    }

    public void setPlayBackListener(PlaybackListener listener) {
        this.listener = listener;
    }

    public void setVolume(float mCurrentVolume) {
    }
}

File: app/src/main/java/com/cyl/musiclake/player/exoplayer/BasePlayer.kt
package com.cyl.musiclake.player.exoplayer

import com.cyl.musiclake.bean.Music
import com.cyl.musiclake.player.playback.PlaybackListener


open class BasePlayer {
    @JvmField
    var mNowPlayingMusic: Music? = null
    @JvmField
    var listener: PlaybackListener? = null
    @JvmField
    var playWhenReady = true

    open fun stop() {}
    open fun release() {}
    open fun start() {}
    open fun pause() {}
    open fun isPlaying(): Boolean {
        return false
    }

    open fun position(): Long {
        return 0
    }

    open fun duration(): Long {
        return 0
    }

    /**
     * 获取标题
     */
    val title: String?
        get() = if (mNowPlayingMusic != null) {
            mNowPlayingMusic!!.title
        } else ""

    /**
     * 获取歌手专辑
     *
     * @return
     */
    val artistName: String?
        get() = if (mNowPlayingMusic != null) {
            mNowPlayingMusic!!.artist
        } else null

    open fun isInitialized(): Boolean {
        return true
    }

    open fun isPrepared(): Boolean {
        return false
    }

    open fun seekTo(ms: Long) {}
    open fun bufferedPercentage(): Int {
        return 0
    }

    open fun getAudioSessionId(): Int {
        return 0
    }

    open fun setDataSource(uri: String?) {}

    open fun setMusicInfo(mNowPlayingMusic: Music?) {
        this.mNowPlayingMusic = mNowPlayingMusic
    }

    open fun setPlayBackListener(listener: PlaybackListener?) {
        this.listener = listener
    }

    open fun setVolume(mCurrentVolume: Float) {}
}